pennylane/devices/default_mixed.py

```diff
@@ -542,7 +589,15 @@ def _apply_operation(self, operation):
         if operation in diagonal_in_z_basis:
             self._apply_diagonal_unitary(matrices, wires)
         else:
-            self._apply_channel(matrices, wires)
+            num_op_wires = len(wires)
+            interface = qml.math.get_interface(*matrices)
+            print(interface)
+            # Use tensordot for Autograd and Numpy if there are more than 2 wires
+            # Use tensordot in any case for more than 7 wires, as einsum does not support this case
+            if (num_op_wires > 2 and interface in {"autograd", None}) or num_op_wires > 7:
```

Comment:
Great :+1:

Reply:
I just noticed a problem with this: This only uses the operation's Kraus matrices to determine the method. However, the device state might lead to the math module dispatching to a different interface (e.g. for non-parametrized operations like a CNOT, the matrices will be "numpy" interface). I think we should take the device state interface into account :thinking: